templates/install_puppet.bat.erb

```diff
@@ -6,6 +6,9 @@ set pid=
 for /f "tokens=2" %%a in ('tasklist /v ^| findstr /c:"%windowTitle%"') do set pid=%%a
 set pid_path=%TEMP%\puppet_agent_upgrade.pid
 
+set environment=
+for /f "delims=" %%i in ('puppet config print --section agent environment') do set environment=%%i
```

Comment:
This is probably the most reasonable approach. Eventually we may be able to guard it depending on which version of the MSI is installing, but currently the MSI always updates the `environment` setting, so we should ensure it uses whatever is configured for the agent section.